core.spec.md

```diff
@@ -0,0 +1,501 @@
+- **Data cores** can then pick up the processed core schema and provide some data-layer service with it. Typically this means serving the schema's API as a GraphQL endpoint, using metadata defined by machinery to inform how it processes operations it receives. However, data cores may perform other tasks described in the core schema, such as routing to backend services, caching commonly-accessed fields and queries, and so on. The term "data core" is intended to capture this multiplicity of possible activities.
+
+# Basic Requirements
+
+Core schemas:
+  1. MUST be valid GraphQL schema documents,
+  2. MUST contain exactly one `SchemaDefinition`, and
+  3. MUST use the {@core} directive on their schema definition to declare any features they reference by using {@core} to reference a [well-formed feature URL](#@core/feature).
+
+The first {@core} directive on the schema MUST reference the core spec itself, i.e. this document.
+
+:::[example](basic.graphql) -- Basic core schema using {@core} and `@example`
+
+## Unspecified directives are passed through by default
+
+Existing schemas likely contain definitions for directives which are not versioned, have no specification document, and are intended mainly to be passed through. This is the default behavior for core schema processors:
+
+```graphql example -- Unspecified directives are passed through
+schema
+  @core(feature: "https://lib.apollo.dev/core/v0.1")
+{
+  query: Query
+}
+
+type SomeType {
+  field: Int @another
+}
+
+# `@another` is unspecified. Core processors will not extract metadata from
+# it, but its definition and all usages within the schema will be exposed
+# in the API.
+directive @another on FIELD
+```
+
+## Renaming core itself
+
+It is possible to rename the core feature itself with the same [`as:`](#@core/as) mechanism used for all features:
+
+```graphql example -- Renaming {@core} to {@coreSchema}
+schema
+  @coreSchema(feature: "https://lib.apollo.dev/core/v0.1", as: "coreSchema")
+  @coreSchema(feature: "https://example.com/example/v1.0")
+{
+  query: Query
+}
+
+type SomeType {
+  field: Int @example
+}
+
+directive @coreSchema(feature: String!, as: String)
+  repeatable on SCHEMA
+directive @example on FIELD
+```
+
+# Directives
+
+##! @core
+
+Declare a core feature present in this schema.
+
+```graphql definition
+directive @core(
+  feature: String!,
+  as: String,
+  export: Boolean! = false)
+  repeatable on SCHEMA
+```
+
+Documents MUST include a definition for the {@core} directive. The provided definition must be *compatible* with the definition above, but may:
+- **Omit arguments** if they are never used in the document,
+- **Omit locations** where the directive never occurs,
```

Comment:
CLARITY: This bullet point seems moot because there is only one location specified.

Reply:
Fixed